src/routes/settings/components/Settings.js

```diff
@@ -0,0 +1,235 @@
+const React = require('react');
+const T = require('prop-types');
+const NavLink = require('react-router-dom').NavLink;
+const StrangeForms = require('strange-forms');
+const IsEmail = require('utils/is-email');
+const { Button, TextField, FormHelperText, Divider, Typography } = require('@material-ui/core');
+const { FormWrapper, TextWrapper, ButtonWrapper } = require('styles/global-components.js');
+
+
+module.exports = class Settings extends StrangeForms(React.Component) {
+
+    static propTypes = {
+        onSubmit: T.func.isRequired,
+        errorMessage: T.string,
+        userDetails: T.shape({
+            firstName: T.string.isRequired,
+            lastName: T.string.isRequired,
+            email: T.string.isRequired
+        })
+    };
+
+    constructor(props) {
+
+        super(props);
+
+        this.state = {
+            firstName: props.userDetails.firstName || '',
+            lastName: props.userDetails.lastName || '',
+            email: props.userDetails.email || '',
+            currentPassword: '',
+            password: '',
+            confirmPassword: '',
+            isBlurred: {
+                email: false,
+                confirmPassword: false
+            }
+        };
+
+        this.strangeForm({
+            fields: ['firstName', 'lastName', 'email', 'currentPassword', 'password', 'confirmPassword'],
+            get: (someProps, field) => this.state[field],
+            act: (field, value) => this.setState({ [field]: value }),
+            getFormValue: {
+                '*': this.getFormValue.bind(this)
+            }
+        });
+
+        this.fieldBlurred = this._fieldBlurred.bind(this);
+        this.showEmailError = this._showEmailError.bind(this);
+        this.showPasswordError = this._showPasswordError.bind(this);
+        this.disableButton = this._disableButton.bind(this);
+        this.submit = this._submit.bind(this);
+    }
+
+    getCheckedValue(e) {
+
+        return e.target.checked;
+    }
+
+    getFormValue(e) {
+
+        return e.target.value || '';
```

Comment:
If the default to `''` isn't necessary, then this doesn't need to be configured at all.  I also try to keep methods that don't need `this` either static (`static getFormValue(e)`) or in a helper (e.g. `internals.getFormValue = (e) => ...`).

Reply:
Good question! I pulled this directly from `Signup.js`, but I can do some testing to see if it works fine sans configuration as a static or helper function.

Reply:
It was propagated from the strange-forms example, which probably originated from me!  I'll make a point to change it.  https://github.com/BigRoomStudios/strange-forms#example

Reply:
What a fun discovery! Do you have an implementation preference between making it `static` vs attaching it to `internals`? We may as well pick your favorite! :)

Reply:
:)

I think my preference would actually be to inline it, come to think of it!  If it's shared across multiple forms, perhaps it would just end-up in a utility.

Reply:
Cool! Right now it looks like it's on `ForgotPassword`, `Login`, `ResetPassword`, `Settings`, and `SignUp`, so maybe a utility makes sense, but at the same time, it's just an itty bitty one-liner. What are your thoughts here?

Reply:
Not a big deal to me either way.  The main thing on my mind is: do any of them need it?

Reply:
^ We chatted about this in-person, but the short answer is "no."

`(e) => e.target.value || ''` is already the default behavior, so in simple cases, `getFormValue()` can be omitted altogether. In other cases, like for a checkbox, we can do something like this:

```js
getFormValue: {
    rememberMe: (e) => e.target.checked,
    '*': (e) => e.target.value || ''
}
```

I thought maybe we could omit the `'*': (e) => e.target.value || ''` case altogether, but that didn't work for me, and it's not bothering me to keep it.

Reply:
Ah, yup, I see in the source it applies the default a little too eagerly and misses the configured-per-field case 😬

Reply:
Should we create an issue in [strange-forms](https://github.com/BigRoomStudios/strange-forms)?

Reply:
Sure!